src/External-contrib/NGFC/SOFC_PR.py

```diff
@@ -0,0 +1,437 @@
+                            ('CO2', 'C3H8'): 0, ('CO2', 'C4H10'): 0,
+                            ('O2', 'H2'): 0, ('O2', 'CO'): 0,
+                            ('O2', 'H2O'): 0, ('O2', 'CO2'): 0,
+                            ('O2', 'O2'): 0, ('O2', 'N2'): 0,
+                            ('O2', 'Ar'): 0, ('O2', 'CH4'): 0,
+                            ('O2', 'C2H6'): 0, ('O2', 'C3H8'): 0,
+                            ('O2', 'C4H10'): 0, ('N2', 'H2'): 0,
+                            ('N2', 'CO'): 0, ('N2', 'H2O'): 0,
+                            ('N2', 'CO2'): 0, ('N2', 'O2'): 0,
+                            ('N2', 'N2'): 0, ('N2', 'Ar'): 0,
+                            ('N2', 'CH4'): 0, ('N2', 'C2H6'): 0,
+                            ('N2', 'C3H8'): 0, ('N2', 'C4H10'): 0,
+                            ('Ar', 'H2'): 0, ('Ar', 'CO'): 0,
+                            ('Ar', 'H2O'): 0, ('Ar', 'CO2'): 0,
+                            ('Ar', 'O2'): 0, ('Ar', 'N2'): 0,
+                            ('Ar', 'Ar'): 0, ('Ar', 'CH4'): 0,
+                            ('Ar', 'C2H6'): 0, ('Ar', 'C3H8'): 0,
+                            ('Ar', 'C4H10'): 0, ('CH4', 'H2'): 0,
+                            ('CH4', 'CO'): 0, ('CH4', 'H2O'): 0,
+                            ('CH4', 'CO2'): 0, ('CH4', 'O2'): 0,
+                            ('CH4', 'N2'): 0, ('CH4', 'Ar'): 0,
+                            ('CH4', 'CH4'): 0, ('CH4', 'C2H6'): 0,
+                            ('CH4', 'C3H8'): 0, ('CH4', 'C4H10'): 0,
+                            ('C2H6', 'H2'): 0, ('C2H6', 'CO'): 0,
+                            ('C2H6', 'H2O'): 0, ('C2H6', 'CO2'): 0,
+                            ('C2H6', 'O2'): 0, ('C2H6', 'N2'): 0,
+                            ('C2H6', 'Ar'): 0, ('C2H6', 'CH4'): 0,
+                            ('C2H6', 'C2H6'): 0, ('C2H6', 'C3H8'): 0,
+                            ('C2H6', 'C4H10'): 0, ('C3H8', 'H2'): 0,
+                            ('C3H8', 'CO'): 0, ('C3H8', 'H2O'): 0,
+                            ('C3H8', 'CO2'): 0, ('C3H8', 'O2'): 0,
+                            ('C3H8', 'N2'): 0, ('C3H8', 'Ar'): 0,
+                            ('C3H8', 'CH4'): 0, ('C3H8', 'C2H6'): 0,
+                            ('C3H8', 'C3H8'): 0, ('C3H8', 'C4H10'): 0,
+                            ('C4H10', 'H2'): 0, ('C4H10', 'CO'): 0,
+                            ('C4H10', 'H2O'): 0, ('C4H10', 'CO2'): 0,
+                            ('C4H10', 'O2'): 0, ('C4H10', 'N2'): 0,
+                            ('C4H10', 'Ar'): 0, ('C4H10', 'CH4'): 0,
+                            ('C4H10', 'C2H6'): 0,
+                            ('C4H10', 'C3H8'): 0,
+                            ('C4H10', 'C4H10'): 0}}}
+
+    if components == 'all':
+        return configuration
+    else:
+        comp_dict = {}
+        for key in components:
+            comp_dict[key] = configuration['components'][key]
+        configuration['components'] = comp_dict
```

Comment:
When trying to create a completely new dict I ran into the issue of module objects (`NIST` and `RPP`) not being copyable. `configuration` is remade each time `get_SOFC_properties()` is called so modifying it shouldn't cause any problems.

Reply:
This issue has also bee fixed in a recent PR - each library now contains a static class of the same name that can be imported instead of the module, which should fix the issue with copying.